src/legacy/core_plugins/metrics/public/components/vis_editor_visualization.js

```diff
@@ -35,74 +35,54 @@ class VisEditorVisualization extends Component {
       panelInterval: 0,
     };
 
-    this.handleMouseUp = this.handleMouseUp.bind(this);
-    this.handleMouseDown = this.handleMouseDown.bind(this);
-    this.onSizeHandleKeyDown = this.onSizeHandleKeyDown.bind(this);
-
     this._visEl = React.createRef();
     this._subscription = null;
   }
 
-  handleMouseDown() {
+  handleMouseDown = () => {
+    window.addEventListener('mouseup', this.handleMouseUp);
     this.setState({ dragging: true });
-  }
+  };
 
-  handleMouseUp() {
+  handleMouseUp = () => {
+    window.removeEventListener('mouseup', this.handleMouseUp);
     this.setState({ dragging: false });
-  }
-
-  componentWillMount() {
-    this.handleMouseMove = (event) => {
-      if (this.state.dragging) {
-        this.setState((prevState) => ({
-          height: Math.max(MIN_CHART_HEIGHT, prevState.height + event.movementY),
-        }));
-      }
-    };
-    window.addEventListener('mousemove', this.handleMouseMove);
-    window.addEventListener('mouseup', this.handleMouseUp);
-  }
+  };
 
-  componentWillUnmount() {
-    window.removeEventListener('mousemove', this.handleMouseMove);
-    window.removeEventListener('mouseup', this.handleMouseUp);
-    if (this._handler) {
-      this._handler.destroy();
-    }
-    if (this._subscription) {
-      this._subscription.unsubscribe();
+  handleMouseMove = (event) => {
+    if (this.state.dragging) {
+      this.setState((prevState) => ({
+        height: Math.max(MIN_CHART_HEIGHT, prevState.height + event.movementY),
+      }));
     }
-  }
-
-  onUpdate = () => {
-    this._handler.update({
-      timeRange: this.props.timeRange,
-    });
   };
 
-  _loadVisualization() {
-    getVisualizeLoader().then(loader => {
-      if (!this._visEl.current) {
-        // In case the visualize loader isn't done before the component is unmounted.
-        return;
-      }
+  async _loadVisualization() {
+    const loader = await getVisualizeLoader();
 
-      this._loader = loader;
-      this._handler = this._loader.embedVisualizationWithSavedObject(this._visEl.current, this.props.savedObj, {
-        uiState: this.props.uiState,
-        listenOnChange: false,
-        timeRange: this.props.timeRange,
-        appState: this.props.appState,
-      });
+    if (!this._visEl.current) {
+      // In case the visualize loader isn't done before the component is unmounted.
+      return;
+    }
 
-      this._subscription = this._handler.data$.subscribe((data) => {
-        this.setPanelInterval(data.visData);
-        this.props.onDataChange(data);
-      });
+    const {
+      uiState,
+      timeRange,
+      appState,
+      savedObj,
+      onDataChange
+    } = this.props;
+
+    this._handler = loader.embedVisualizationWithSavedObject(this._visEl.current, savedObj, {
+      listenOnChange: false,
+      uiState,
+      timeRange,
+      appState,
+    });
 
-      if (this._handlerUpdateHasAlreadyBeenTriggered) {
```

Comment:
It seems like this was a workaround to make sure the update is correctly propagated even if the visualization hasn't loaded yet. Why was it removed? Is this not necessary anymore?

Reply:
Seems like it was a redundant update like the others in the file!
The `timeRange` which the `handler.update` expect is actual because it is taken from props. Why it was updated with the same value right after initialize?..
So I do not see any necessity in this update..
If I'm wrong, please correct me @timroes

Reply:
@flash1293  about **_handlerUpdateHasAlreadyBeenTriggered**. Not sure that it was a good idea to add this type of handler. If removing will cause some regression it should be fixed in a different way.

Reply:
I tried to trigger a bug based on this, but it seems like it works fine, at least at the moment. We should keep it in mind if something breaks based on this.

Reply:
This workaround was introduced because of https://github.com/elastic/kibana/issues/21426, @sulemanof could we check whether this re-introduced the bug? If everything works IMO we can go with the current solution. We should keep it in mind though - the current solution is not 100% waterproof.

Reply:
@flash1293 the error can't be reproduced since it have a check:
```` 
componentDidUpdate(prevProps) {
    if (this._handler && !isEqual(this.props.timeRange, prevProps.timeRange)) {
      this._handler.update({
        timeRange: this.props.timeRange,
      });
    }
 }

Reply:
@sulemanof The check prevents it from crashing, but does it also apply the time range correctly? If this is the case, it LGTM 👍